Migrating from Jandex maven plugin 1.1.1 to 3.1.3 fails to create intermediate directories when generating an index

Consider the following jandex plugin configuration:

```xml
          <execution>
            <id>make-index-country</id>
            <goals>
              <goal>jandex</goal>
            </goals>
            <!-- phase is 'process-classes by default' -->
            <phase>pre-integration-test</phase>
            <configuration>
              <indexName>country-index/custom.idx</indexName>
              <fileSets>
                <fileSet>
                  <directory>${testClassesDirectory}/data/portabletype</directory>
                  <includes>
                    <include>Country.class</include>
                  </includes>
                </fileSet>
              </fileSets>
            </configuration>
          </execution>
```

This fails with the following:

```
Caused by: java.nio.file.NoSuchFileException: /Users/rlubke/files/coh/dev/release/coherence-v14.1.1.2206/prj/test/functional/io/target/classes/META-INF/address-index/custom.idx
    at sun.nio.fs.UnixException.translateToIOException (UnixException.java:92)
    at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:111)
    at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:116)
    at sun.nio.fs.UnixFileSystemProvider.newFileChannel (UnixFileSystemProvider.java:182)
    at java.nio.channels.FileChannel.open (FileChannel.java:292)
    at java.nio.channels.FileChannel.open (FileChannel.java:345)
    at org.codehaus.plexus.util.io.CachingOutputStream.<init> (CachingOutputStream.java:57)
    at org.codehaus.plexus.util.io.CachingOutputStream.<init> (CachingOutputStream.java:51)
    at org.codehaus.plexus.util.io.CachingOutputStream.<init> (CachingOutputStream.java:46)
    at org.jboss.jandex.maven.JandexGoal.execute (JandexGoal.java:176)

```

Workaround:  use the antrun plugin to create the target directories prior to generating the indexes.

--- a/maven-plugin/src/main/java/org/jboss/jandex/maven/JandexGoal.java
+++ b/maven-plugin/src/main/java/org/jboss/jandex/maven/JandexGoal.java
@@ -172,7 +172,7 @@ public class JandexGoal extends AbstractMojo {
         File indexFile = new File(indexDir, indexName);
         getLog().info("Saving Jandex index: " + indexFile);
         try {
-            Files.createDirectories(indexDir.toPath());
+            Files.createDirectories(indexFile.toPath().getParent());
             try (OutputStream out = new CachingOutputStream(indexFile)) {
                 IndexWriter writer = new IndexWriter(out);
                 if (indexVersion != null) {

